eqcorrscan/utils/plotting.py

```diff
@@ -2186,6 +2186,131 @@ def subspace_fc_plot(detector, stachans, **kwargs):
     return fig
 
 
+@additional_docstring(plotting_kwargs=plotting_kwargs)
+def mapplot(events, bgcolor='#909090', method='depth', cpalette='jet_r',
+            s=1, lw=1, marker=',', **kwargs):
+    """
+    Plot seismicity in a 2D map with two cross section along latitude and
+    longitude.
+
+    :type events: list or 'obspy.core.event.catalog.Catalog'
+    :param events:
+        list of one tuple per event of (lat, long, depth(optional)) with
+        down positive or obspy catalog.
+    :type bgcolor: string
+    :param bgcolor: all name or RGB code that acceptable in matplotlib.
+    :type method: string
+    :param method:
+        make color pallete according to thrid part of area 'depth' or
+        occouring time 'time' or occouring sequence 'sequence'.
+    :type cpalette: string
+    :param cpalette:
+        color palette for drawing events. acceptable with matplotlib.
+    :type s: scalar or array_like, shape (n, ), default: None
+    :param s:
+        The marker size in points**2.
+        Default is rcParams['lines.markersize'] ** 2.
+    :type lw: scalar or array_like, default: None
+    :param lw:
+        The linewidth of the marker edges.
+        If None, defaults to rcParams lines.linewidth.
+    :type marker: string
+    :param marker:
+        The marker style. *marker* can be either an instance of the class
+        or the text shorthand for a particular marker.
+        Defaults to '.'
+    {plotting_kwargs}
+
+    :returns: :class:`matplotlib.figure.Figure`
+
+    .. image:: ../doc/plots/mapplot_depth.png
+    .. image:: ../doc/plots/mapplot_time.png
+    .. image:: ../doc/plots/mapplot_sequrnce.png
```

Comment:
If giving these examples, provide the code-snippet used to create them, or at least the argument differences to create them